bokeh/core/property/bases.py

```diff
@@ -0,0 +1,378 @@
+''' Provide base classes for the Bokeh property system.
+
+.. note::
+    These classes form part of the very low-level machinery that implements
+    the Bokeh model and property system. It is unlikely that any of these
+    classes or their methods will be applicable to any standard usage or to
+    anyone who is not directly developing on Bokeh's own infrastructure.
+
+'''
+from __future__ import absolute_import
+
+import logging
+logger = logging.getLogger(__name__)
+
+from copy import copy
+import types
+
+from six import string_types
+
+from ...util.string import nice_join
+from .containers import PropertyValueList, PropertyValueDict
+from .descriptors import BasicPropertyDescriptor
+from .factory import PropertyFactory
+
+class DeserializationError(Exception):
+    pass
+
+class Property(PropertyFactory):
```

Comment:
yeah, you really have to rename PropertyFactory now, properties are their own factory? ;-)